ftd/src/interpreter/test.rs

```diff
@@ -258,3 +258,41 @@ fn evalexpr_test() {
         }
             */
 }
+
+#[test]
+fn test_extract_kwargs() {
+    let doc = interpret_helper(
+        "foo",
+        r#"-- component foo:
+    kw-args data:
+
+    -- ftd.text: Hello world
+
+    -- end: foo
+
+    -- foo:
+    bar: Hello
+    baz: World
+"#,
+    )
+    .unwrap_or_else(|e| panic!("{:?}", e));
+
+    let tdoc = doc.tdoc();
+
+    let instructions = doc.get_instructions("foo#foo");
+    let instruction = instructions.first().unwrap();
+    let kwargs = instruction.get_kwargs("data", &tdoc).unwrap().unwrap();
+    // TODO: resolve value based on kwarg kind
+    let data: std::collections::BTreeMap<String, String> = kwargs
+        .iter()
+        .map(|(name, property_value)| {
+            (
+                name.to_string(),
+                property_value.to_value().get_string_data().unwrap(),
+            )
+        })
+        .collect();
```

Comment:
Can you make this two lines?

Reply:
the first few lines are boilerplate code. Do you mean the `let data = ..` line?

Reply:
I mean can you take lines I have highlighted and make it two lines. Remove the boilerplace code.